alibi/explainers/partial_dependence.py

```diff
@@ -0,0 +1,1294 @@
+            f0, f1 = feature_names
+
+            if (not _is_categorical(f0)) and (not _is_categorical(f1)):
+                _ = _plot_two_pd_num_num(exp=exp,
+                                         feature=features,
+                                         target_idx=target_idx,
+                                         levels=levels,
+                                         ax=ax_ravel,
+                                         pd_num_num_kw=pd_num_num_kw)
+
+            elif _is_categorical(f0) and _is_categorical(f1):
+                _ = _plot_two_pd_cat_cat(exp=exp,
+                                         feature=features,
+                                         target_idx=target_idx,
+                                         ax=ax_ravel,
+                                         pd_cat_cat_kw=pd_cat_cat_kw)
+
+            else:
+                _ = _plot_two_pd_num_cat(exp=exp,
+                                         feature=features,
+                                         target_idx=target_idx,
+                                         ax=ax_ravel,
+                                         pd_num_cat_kw=pd_num_cat_kw)
+
+        else:
+            if _is_categorical(feature_names):
+                _ = _plot_one_pd_cat(exp=exp,
+                                     feature=features,
+                                     target_idx=target_idx,
+                                     centered=centered,
+                                     n_ice=n_ice,
+                                     ax=ax_ravel,
+                                     pd_cat_kw=pd_cat_kw,
+                                     ice_cat_kw=ice_cat_kw,
+                                     seed=seed)
+            else:
+                _ = _plot_one_pd_num(exp=exp,
+                                     feature=features,
+                                     target_idx=target_idx,
+                                     centered=centered,
+                                     n_ice=n_ice,
+                                     ax=ax_ravel,
+                                     pd_num_kw=pd_num_kw,
+                                     ice_num_kw=ice_num_kw,
+                                     seed=seed)
+
+    fig.set(**fig_kw)
+    return axes
+
+
+def _sample_ice(ice_values: np.ndarray,
+                n_ice: Union[str, int, List[int]],
+                seed: Optional[int] = None) -> np.ndarray:
+    """
+    Samples ice_values based on the n_ice argument.
+
+    Parameters
+    ----------
+    ice_values
+        Array of ice_values of dimension (N, V), where N is the number of instances in the reference dataset,
+        and V is the number of feature values where the PD is computed.
+    n_ice
+        See :py:meth:`alibi.explainers.partial_dependence.plot_pd`.
+    seed
+        Seed to be used for sampling.
+    """
+    np.random.seed(seed)
+
+    if n_ice == 'all':
+        return ice_values
+
+    _, N = ice_values.shape
```

Comment:
Nitpick: in the docstring the shape is written as `(N, V)` but here we use the variable `N` for the second entry.